pkg/controller/kubelet-config/helpers.go

```diff
@@ -110,13 +136,85 @@ func updateOriginalKubeConfigwithNodeConfig(node *osev1.Node, originalKubeletCon
 		originalKubeletConfig.NodeStatusUpdateFrequency = metav1.Duration{Duration: osev1.LowNodeStatusUpdateFrequency}
 	case osev1.DefaultUpdateDefaultReaction:
 		originalKubeletConfig.NodeStatusUpdateFrequency = metav1.Duration{Duration: osev1.DefaultNodeStatusUpdateFrequency}
+	case emptyInput:
+		return nil
 	default:
 		return fmt.Errorf("unknown worker latency profile type found %v, failed to update the original kubelet configuration", node.Spec.WorkerLatencyProfile)
 	}
 	// The kubelet configuration can be updated based on the cgroupmode as well here.
 	return nil
 }
 
+// isTechPreviewNoUpgradeEnabled returns a boolean accordingly if a feature is enabled
+func isTechPreviewNoUpgradeEnabled(features *osev1.FeatureGate, feature string) bool {
+	if features == nil {
+		return false
+	}
+	if features.Spec.FeatureGateSelection.FeatureSet == osev1.TechPreviewNoUpgrade {
+		featureSet, ok := osev1.FeatureSets[osev1.TechPreviewNoUpgrade]
+		if !ok {
+			return false
+		}
+		for _, val := range featureSet.Enabled {
+			if val == feature {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// updateMachineConfigwithCgroup updates the Machine Config object based on the cgroup mode present in the Config Node resource.
+func updateMachineConfigwithCgroup(node *osev1.Node, mc *mcfgv1.MachineConfig) error {
+	if node == nil {
+		return fmt.Errorf("node configuration not found, failed to update the machine config with the cgroup information")
+	}
+	if reflect.DeepEqual(node.Spec, osev1.NodeSpec{}) {
+		return fmt.Errorf("empty config node resource spec found")
+	}
+	if mc == nil || reflect.DeepEqual(mc.Spec, mcfgv1.MachineConfigSpec{}) {
+		return fmt.Errorf("machine config not found, failed to update the machine config with the cgroup information")
+	}
+	// updating the Machine Config resource with the relevant cgroup config
+	var (
+		kernelArgsv1       = []string{"systemd.unified_cgroup_hierarchy=0"}
```

Comment:
As a follow up, let's see if we can also add ` systemd.legacy_systemd_cgroup_controller=` for v1.

Reply:
Addressed, Updated the args slice with the new parameter and tested on a cluster, observed that the nodes were successfully updated without any issue.